modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/impl/WFSContentDataAccess.java

```diff
@@ -197,8 +215,27 @@ public FeatureSource<FeatureType, Feature> getFeatureSource(Name typeName) throw
         return contentComplexFeatureSource;
     }
 
+    private void deleteTemporaryCache() {
+        if (this.schemaParser.getResolver() != null) {
+            SchemaCache cache = this.schemaParser.getResolver().getCache();
+            if (cache != null) {
+                try {
+                    FileUtils.deleteDirectory(cache.getDirectory());
+                } catch (IOException e) {
+                    LOGGER.log(
+                            Level.SEVERE,
+                            "Failed to delete temporary directory used for wfs schema cache.",
+                            e);
+                }
+            }
+        }
+    }
+
     @Override
     public void dispose() {
+        if (this.cacheLocation == null && this.schemaParser != null) {
```

Comment:
Same as above.

Reply:
cacheLocation == null indicates that we don't have a user-provided schema folder, so it should be safe to delete the folder.